### ISSUE TITLE
Question: which cake units are used?

### ISSUE THREAD

**Issue by lcn2** [ioccc-src/winner#12]
I (Landon) plan to bake the cake sometime this northern hemisphere winter/spring 2022:

> [let them eat Double-layered Chocolate Fudge Cake](https://www.ioccc.org/2020/ferguson1/chocolate-cake.html) :-)

I have this question, @xexyl: The above URL mentions:

> For example a US cup is not the same as Canadian cup and a metric cup differs too. There’s also a difference for teaspoons and tablespoons: there are metric, UK and US teaspoons and tablespoons. There’s both a UK and US ounce also.

The above link does not appear to state, at least explicitly, which units are being used in the recipe.  For example, when it refers to cups or teaspoons or tablespoons, which units are being referenced?  UK or US teaspoons / tablespoons?  US or Canadian cups? etc.  I.e., unless it says otherwise, which measurement system is reference in the recipe?

**Comment by xexyl** [comment]
It uses US measurements. It was noted elsewhere in the file but maybe I should remove that paragraph and just explicitly state it's the US measurements - and give the resource that can convert in case the person is not in the US?

I look forward to seeing pictures of the cake! Like I said I might have a slight emendation on the recipe (with the icing) but I won't have a confirmation until my birthday (it does seem like it: the idea is to use pure vanilla extract rather than imitation vanilla).

Anyway to recap:

- It uses US measurements.
- Should I maybe remove that block of text and just explicitly state US (I think I have) and then give a resource in case the person is not in the US? Something tells me that might be clearer?

Of course if I do have to update it I have to update the Enigma one too which means running my Enigma machine again :) I will also try updating these entries as well as there are some fixes e.g. with the Makefile for Snake.

**Comment by lcn2** [comment]
> Btw out of curiosity: Let's say that my entries did not win. Normally you shred all papers etc. for the entries that do not win. In this case would you have done that with the recipe? I understand that might go against your 'rule' about telling how many entries are entered etc. but in case you're willing to say I'm very interested.

We would have not consider the recipe of a non-winner, as judging is based on the code, not things referenced by the code.

**Comment by lcn2** [comment]
> It uses US measurements. It was noted elsewhere in the file but maybe I should remove that paragraph and just explicitly state it's the US measurements - and give the resource that can convert in case the person is not in the US?
> 
> I look forward to seeing pictures of the cake! Like I said I might have a slight emendation on the recipe (with the icing) but I won't have a confirmation until my birthday (it does seem like it: the idea is to use pure vanilla extract rather than imitation vanilla).
> 
> Anyway to recap:
> 
> * It uses US measurements.
> * Should I maybe remove that block of text and just explicitly state US (I think I have) and then give a resource in case the person is not in the US? Something tells me that might be clearer?
> 
> Of course if I do have to update it I have to update the Enigma one too which means running my Enigma machine again :) I will also try updating these entries as well as there are some fixes e.g. with the Makefile for Snake.

You might consider a pull request on two 2020 cake related recipes, @xexyl

**Comment by xexyl** [comment]
﻿
﻿
﻿
I should have been a bit clearer. What I meant is that the losing entries are shredded. Let’s say I only submitted the entries that didn’t win (but happened to have the same recipe). Since you the judges love chocolate (and who can blame you ? :) ) would you have got rid of the recipe?

I still hope to one day see you tweet about it though! I am happy you stayed safe with judging but I was a bit sad that I didn’t see any pictures. As I am sure your cook would have made it. 

I will get back to the other things tomorrow. I have a commit lined up for the answers file that complements Ilya’s changes to my original code. 

I do live in the Pacific time zone but my sleep is horrible and I go to bed early as I wake up early irrespective of when I go to sleep. 

Last I knew you were in the Bay Area so I hope you enjoy the rest of your evening!

Cheers.

**Comment by xexyl** [comment]
﻿
I will wait until I have verified that the problem with the icing was indeed the imitation vanilla. It almost certainly was that. There was a time where the grocer only had imitation vanilla and the icing didn’t come out well. The last time the cake was made it was with pure vanilla extract and it turned out fine. 

Usually the cake for my birthday is made on my birthday but because it takes a lot of time and because my cousin will be here the day before (since it falls on Monday this year - though she will be here in the evening on my birthday too) it will be made two days before so this year 12 February. 

In all likelihood this is the problem and if so I will have reason to update the file (and make it clearer that it’s US measurements) and at the same time I will update the other files.

The Makefile for Snake especially needs to be fixed as after you changed the markdown extension to be text (which Simon did warn us in 2018 that this might happen eventually) the scripts I included are broken as you have the markdown files in the dependency of the make target. This in turn means that the script that has a bunch of preconfigured gameplay options doesn’t work. 

But I have had a lot going on since the end of 2020. I thought I had done a pull request but I still am fairly new to GitHub (all my previous experience with git is private servers) so I did it wrong (as you informed me on Twitter - thank you for that!). 

I do have a question though (I know it took a while to get to it and I am sorry about that): I know originally there was a problem with the html files not being rendered on GitHub so in my commit I generated the html files locally and then committed those (making sure to add the footer and header that belongs with the 2020 entries). Is this still necessary though?

Thank you!
